compiler/rustc_macros/src/diagnostics/fluent.rs

```diff
@@ -205,11 +205,23 @@ pub(crate) fn fluent_messages(input: proc_macro::TokenStream) -> proc_macro::Tok
                 // The last case we error about above, but we want to fall back gracefully
                 // so that only the error is being emitted and not also one about the macro
                 // failing.
-                let snake_name = Ident::new(
-                    // FIXME: should probably trim prefix, not replace all occurrences
-                    &name.replace('-', "_").replace(&format!("{}_", res.ident), ""),
-                    span,
-                );
+                let crate_prefix = format!("{}_", res.ident);
+
+                let snake_name = name.replace('-', "_");
+                let snake_name = match snake_name.strip_prefix(&crate_prefix) {
+                    Some(rest) => Ident::new(rest, span),
+                    None => {
+                        Diagnostic::spanned(
```

Comment:
This is great, could you add a failing test in `src/test/ui-fulldeps/fluent-messages/test.rs`?

Reply:
I suspect this test will be failing anyway given the `Span` changes, so you would have to bless it anyway.

Reply:
Yeah, had to update a couple tests, should be good now.